### PR TITLE
Tc21 add review

### PR DESCRIPTION
**Pull request by HadeerAlaa1**
[Kindly check pipeline run 
](https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/8094504574)

Dear @MahmoudElSharkawyGS  for the pipeline link i already added here, and for the conflicts i will double check

### CODE REVIEW

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:1]: Why delete all empty lines between method here?

File: src/main/java/com/gizasystems/automationexercise/pages/NavigationBar.java [hunk 1]
```diff
@@ -31,10 +33,24 @@ public NavigationBar clickOnDeleteAccountLink() {
         return this;
     }
 
+    @Step("Click on Products link")
+    public NavigationBar clickOnProductsLink() {
+        driver.element().click(products);
+        return this;
+    }
+    @Step("Handle ADS")
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/NavigationBar.java:41]: Please add empty line here

File: src/main/java/com/gizasystems/automationexercise/pages/NavigationBar.java [hunk 2]
```diff
@@ -31,10 +33,24 @@ public NavigationBar clickOnDeleteAccountLink() {
...
+    public NavigationBar handleAd(){
+        Actions actions = new Actions(driver.getDriver());
+        actions.pause(1000).doubleClick().perform();
+        return this;
+    }
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/NavigationBar.java:46]: Why handling the add in the navigation bar page?
Also, I believe we already have a method for this (please check the utils package).

File: src/main/java/com/gizasystems/automationexercise/pages/NavigationBar.java [hunk 3]
```diff
@@ -31,10 +33,24 @@ public NavigationBar clickOnDeleteAccountLink() {
...
     //////////////////// Validations \\\\\\\\\\\\\\\\\\\\
     @Step("Validate the Logged In User")
     public NavigationBar validateTheLoggedInUser(String expectedUser) {
         driver.verifyThat().element(loggedInUser_link).text().isEqualTo("Logged in as " + expectedUser).perform();
         return this;
     }
+
+
+
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/NavigationBar.java:55]: Please delete extra empty lines.

File: src/main/java/com/gizasystems/automationexercise/pages/ProductsPage.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    // Locators
+    private final By searchProduct = By.id("search_product");
+    private final By viewProduct = By.xpath("(//i[@class='fa fa-plus-square'])[1]");
+    private final By writeReviewSection = By.xpath("//li[@class='active']");
+    private final By reviewerName_input =By.id("name");
+    private final By reviewerEmail_input =By.id("email");
+    private final By reviewText_input = By.name("review");
+    private final By submitBtn = By.id("button-review");
+    private final By reviewSuccessAlert = By.cssSelector("div.alert-success.alert > span[style=\"font-size: 20px;\"]");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/ProductsPage.java:19]: Let's check this locator together as I believe we can enhance it.

File: src/test/java/com/gizasystems/automationexercise/tests/reviews/AddReviewTests.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                .AddReviewOnProduct(testData.getTestData("UserName"),
+                        testData.getTestData("UserMail.GuiTimeStamp") + timeStamp + "@gizasystems.com",
+                        reviewTestData.getTestData("ReviewText"))
+                .validatethatReviewSuccessAlertIsDisplayed();
+    }
+
+    //////////////////// Configurations \\\\\\\\\\\\\\\\\\\\
+    @BeforeClass
+    public void beforeClass() {
+        testData = new SHAFT.TestData.JSON("src/test/resources/testDataFiles/RegisterUser.json");
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/reviews/AddReviewTests.java:39]: Only one test data file needed to maintain the 1 to 1 relationship between testdata file and test classes.

File: src/test/java/com/gizasystems/automationexercise/tests/reviews/AddReviewTests.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+public class AddReviewTests {
+    private SHAFT.GUI.WebDriver driver;
+    private SHAFT.TestData.JSON testData,reviewTestData;
+    private String timeStamp = String.valueOf(System.currentTimeMillis());
+
+    @Test(description = "Add review on product")
+    public void addReview(){
+        new NavigationBar(driver).clickOnProductsLink().handleAd();
+        new ProductsPage(driver).validateOnVisibilityOfProductsPage();
+        new ProductsPage(driver).clickOnViewProduct()
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/reviews/AddReviewTests.java:28]: Why you need a new initialization while both actions on the same page; We need to maintain the fluency please.

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
Please make sure to format your code using (alt + ctrl + L)

**Comment by MahmoudElSharkawyGS** [comment]
@HadeerAlaa1 Any updates regarding tis PR please?

**Comment by MahmoudElSharkawyGS** [comment]
@HadeerAlaa1 Thank you for your efforts.
- Please resolve the conflicts
- Please make to execute the pipeline after every change and share the pipeline execution link here

**Comment by MahmoudElSharkawyGS** [comment]
@HadeerAlaa1 the conflicts still not resolved

**Comment by HadeerAlaa1** [comment]
Please find the pipline link attached here, https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/8094504574

**Comment by HadeerAlaa1** [comment]
@MahmoudElSharkawyGS  kindly check my PR and here is the pipeline link 
 https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/8098460712

**Pull request closed without merging**